Automattic/woocommerce-payments
# Update payment method selection UI and fee display

Fixes #3300
Reference paJDYF-2rk-p2

### Changes proposed in this Pull Request

This PR applies the changes described in paJDYF-2rk-p2 to the UPE workflow:

## All screens
- Shows detailed fee breakdown for the PM
- Adds a 1500ms delay and loading animation to the checkbox check event
- Displays activation modal after checkbox delay if the PM has requirements

## Payment settings
- Removes the payment method selector modal and adds checkboxes directly to the payment method list
- Removes the delete button and moves the action to the checkbox uncheck event.
- Delete modal shows for disabling an already active PM
- Moves the activation modal display to the checkbox check event.
- Lists all the available methods instead showing only the enabled methods
- Removes the add payment method button

## UPE onboarding & UPE setup screens
- All the checkboxes start unchecked now, to display the activation modal when any requirements are present for the PM.

---

### Screenshots

## Payment settings (http://localhost:8082/wp-admin/admin.php?page=wc-settings&tab=checkout&section=woocommerce_payments)

![image](https://user-images.githubusercontent.com/3295/142390187-72bc3476-a618-4e6e-a6a5-17aed7617be2.png)

![image](https://user-images.githubusercontent.com/3295/142390302-c15ecf7d-83fe-458e-bfa4-41db912b0afa.png)

![image](https://user-images.githubusercontent.com/3295/142394435-c8339351-f865-45dd-a0e2-51bbb7340ad2.png)



## UPE Onboarding (http://localhost:8082/wp-admin/admin.php?page=wc-admin&task=woocommerce-payments--additional-payment-methods)

![image](https://user-images.githubusercontent.com/3295/142390814-502b227f-32e9-4658-97ab-f98685cdaed4.png)

![image](https://user-images.githubusercontent.com/3295/142390724-3cdeb4ae-a1a7-4bde-984d-863c19075f9e.png)



## UPE Setup (http://localhost:8082/wp-admin/admin.php?page=wc-admin&task=woocommerce-payments--additional-payment-methods)

![image](https://user-images.githubusercontent.com/3295/142392250-9dbeada4-7e64-4f7f-a719-ec4577b7bd06.png)

![image](https://user-images.githubusercontent.com/3295/142392355-530767d5-815c-40ce-9f01-dce4f100b64b.png)


---

### Testing instructions

* All tests should pass
* Check if the screens are in sync with the flow described in paJDYF-2rk-p2 (and it's comments)
* You can check how to simulate different statuses for PM's in the testing instructions of  https://github.com/Automattic/woocommerce-payments/pull/3093 
* You can reach to the PM setup screen via returning this setting `false`: https://github.com/Automattic/woocommerce-payments/blob/de28a8d26f0a93507700fdaae78b18f2d8d8f6b8/includes/admin/class-wc-payments-admin.php#L326

-------------------


**Post merge**
